baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -430,16 +439,150 @@ def hashes_comply(s1, s2):
+                            + (artifact.get("snapshot_version", None),)
+                        )
                     ),
                     iter(snapshot_array),
                 ),
-                {"url": None, "block_hash": None},
+                None,
             )
 
-            self.config["snapshots"][name] = snapshot_metadata
+        def get_artifact_node_version(artifact):
+            version = artifact["tezos_version"]["version"]
+            # there seem to be some inconsistency with that field in different providers
+            # so the only thing we check is if it's a string
+            additional_info = version["additional_info"]
+            return (
+                version["major"],
+                version["minor"],
+                None if type(additional_info) == str else additional_info["rc"],
+            )
+
+        def compose_pred(*preds):
+            return reduce(
+                lambda acc, x: lambda major, minor, rc, snapshot_version: acc(
+                    major, minor, rc, snapshot_version
+                )
+                and x(major, minor, rc, snapshot_version),
+                preds,
+            )
+
+        def sum_pred(*preds):
+            return reduce(
+                lambda acc, x: lambda major, minor, rc, snapshot_version: acc(
+                    major, minor, rc, snapshot_version
+                )
+                or x(major, minor, rc, snapshot_version),
+                preds,
+            )
+
+        node_version = get_node_version()
+        major_version, minor_version, rc_version = node_version
+
+        exact_version_pred = (
+            lambda major, minor, rc, snapshot_version: node_version
+            == (
+                major,
+                minor,
+                rc,
+            )
+        )
+
+        exact_major_version_pred = (
+            lambda major, minor, rc, snapshot_version: major_version == major
+        )
+
+        exact_minor_version_pred = (
+            lambda major, minor, rc, snapshot_version: minor_version == minor
+        )
+
+        less_minor_version_pred = (
+            lambda major, minor, rc, snapshot_version: minor_version > minor
+        )
+
+        exact_rc_version_pred = (
+            lambda major, minor, rc, snapshot_version: rc_version == rc
+        )
+
+        less_rc_version_pred = (
+            lambda major, minor, rc, snapshot_version: rc and rc_version > rc
+        )
+
+        compatible_version_pred = (
+            # it could happen that `snapshot_version` field is not supplied by provider
+            # e.g. marigold snapshots don't supply it
+            lambda major, minor, rc, snapshot_version: snapshot_version
+            and compatible_snapshot_version == snapshot_version
+        )
+
+        no_rc_on_stable_pred = lambda major, minor, rc, snapshot_version: not (
+            rc and rc_version is None
+        )
+
+        every_pred = lambda major, minor, rc, snapshot_version: True
```

Comment:
Do we really need this predicate? :thinking: 

IMO it's better to return `None` and say that we didn't find a compatible snapshot rather than picking the one which is mist likely incompatible and fail during the import

Or am I missing something and it won't work like this?